aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -93,7 +97,23 @@ struct DispatchKeyExtractor final {
     return typeSetToDispatchKey_(type_set);
   }
 
+  c10::optional<size_t> getScalarTypeIndex() const {
+    return scalar_type_index_;
+  }
+
 private:
+  static c10::optional<size_t> getOptionalIndexForArgType_(
+    TypePtr type,
```

Comment:
`const TypePtr&` to avoid the refcount bump is a bit faster, TypePtr is a shared_ptr